core/property/registry.go

```diff
@@ -2,31 +2,66 @@ package property
 
 import (
 	"fmt"
-	"sync"
-
+	"github.com/TIBCOSoftware/flogo-lib/config"
 	"github.com/TIBCOSoftware/flogo-lib/core/data"
 	"github.com/TIBCOSoftware/flogo-lib/logger"
+	"reflect"
+	"regexp"
+	"sync"
 )
 
 var (
-	props = make(map[string]string)
-	mut   = sync.RWMutex{}
+	props    = make(map[string]interface{})
+	mut      = sync.RWMutex{}
+	regex    = regexp.MustCompile(config.GetPropertyDelimiterFormat())
+	resolver PropertyValueResolver
 )
 
+func init() {
+	logger.Debugf("Registering environment variable value resolver")
+	RegisterValueResolver(&data.EnvVarResolver{})
+}
+
+// Resolve value sourced from Enviornment variable or any other configuration management services
+type PropertyValueResolver interface {
```

Comment:
I would shorten the name of this interface to:

type Resolver interface{}

Since it lives in property package. It will read pretty well like:

property.Resolver